b1ddi/schema_dns_data_record_rdata.go

```diff
@@ -0,0 +1,106 @@
+package b1ddi
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+var (
+	ParseError = "unable to parse key '%s': %v"
+)
+
+/*
+	updateDataRecordRData helps convert rDATA record fields of type integer from string value
+    Introduced to fix issues where terraform converts integer values to string in rendered config
+*/
+func updateDataRecordRData(d interface{}, recordType string) (interface{}, diag.Diagnostics) {
+	if d == nil {
+		return nil, nil
+	}
+
+	var diags diag.Diagnostics
+
+	in := d.(map[string]interface{})
+	switch recordType {
+	case "CAA":
+		if val, ok := in["flags"]; ok {
+			i, err := strconv.Atoi(val.(string))
+			if err != nil {
+				diags = append(diags, diag.FromErr(errors.New(fmt.Sprintf(ParseError, "flags", err)))...)
+			} else {
+				in["flags"] = i
+			}
+		}
```

Comment:
Any reason to not use a helper function for this instead of duplicating the same block?